lib/manageiq/rpm_build/generate_ansible_venv.rb

```diff
@@ -0,0 +1,78 @@
+require 'fileutils'
+
+module ManageIQ
+  module RPMBuild
+    class GenerateAnsibleVenv
+      include Helper
+
+      attr_reader :current_env, :manifest_dir, :venv_dir
+
+      def initialize
+        where_am_i
+
+        @manifest_dir = MANIFEST_DIR
+        @venv_dir     = BUILD_DIR.join("manageiq-ansible-venv")
+
+        @venv_path    = @venv_dir.join("venv")
+        @py_version   = "3.8"
+      end
+
+      def populate
+        where_am_i
+
+        FileUtils.mkdir_p(venv_dir)
+        Dir.chdir(venv_dir) do
+          install_python_system_packages
+          create_venv
+          install_venv_packages
+          generate_manifest
+          scrub
+        end
+      end
+
+      private
+
+      def install_python_system_packages
+        where_am_i
+
+        shell_cmd("pip#{@py_version} install virtualenv psutil==5.6.6")
+      end
+
+      def create_venv
+        where_am_i
+
+        FileUtils.mkdir_p(venv_dir)
+        shell_cmd("virtualenv --system-site-packages #{@venv_path}")
+      end
+
+      def install_venv_packages
+        where_am_i
+
+        shell_cmd_in_venv("pip#{@py_version} install --no-compile -r #{CONFIG_DIR.join("requirements.txt")}")
+      end
+
+      def generate_manifest
+        where_am_i
+
+        shell_cmd_in_venv("pip#{@py_version} install pip-licenses")
+        shell_cmd_in_venv("pip-licenses --from=mixed --format=csv --output-file=#{venv_dir.join("ansible_venv_manifest.csv")}")
+        shell_cmd_in_venv("pip#{@py_version} uninstall -y pip-licenses PTable")
+      end
+
+      def scrub
+        where_am_i
+
+        # Remove unneeded files
+        Dir.chdir(@venv_path) do
+          FileUtils.rm_rf(Dir.glob("share/doc/*"))
+        end
+      end
+
+      def shell_cmd_in_venv(cmd)
```

Comment:
This method is essentially a copy of shell_cmd but sourcing the venv has to happen in the same command, so this method is an extension that includes that.  I could have used shell_cmd and duplicate the sourcing, but then the printout is really nasty, whereas this just says `---> (venv) <cmd>`

Reply:
I'd probably be ok moving this to the helper so at least both shell_cmd are in the same place and change in one would alert the author to also change the other.  It's up to you.

Reply:
Ok I moved this to the helper.